File: manual/data/java/tests/test_Schema_validate.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_Schema_validate {

    @Test
    void test_valid_data() {
        function_Schema_validate validator = new function_Schema_validate();
        String result = validator.validate("Alice", "30", "New York");
        assertEquals("Valid data: name = Alice, age = 30, city = New York", result);
    }

    @Test
    void test_invalid_age() {
        function_Schema_validate validator = new function_Schema_validate();
        String result = validator.validate("Alice", "15", "New York");
        assertEquals("Invalid data: One or more fields are incorrect", result);
    }

    @Test
    void test_invalid_city() {
        function_Schema_validate validator = new function_Schema_validate();
        String result = validator.validate("Bob", "25", "12345");
        assertEquals("Invalid data: One or more fields are incorrect", result);
    }

    @Test
    void test_missing_city() {
        function_Schema_validate validator = new function_Schema_validate();
        String result = validator.validate("Charlie", "28", "");
        assertEquals("Invalid data: One or more fields are incorrect", result);
    }

    @Test
    void test_invalid_name() {
        function_Schema_validate validator = new function_Schema_validate();
        String result = validator.validate("", "45", "London");
        assertEquals("Invalid data: One or more fields are incorrect", result);
    }
}

File: manual/data/python/tests/test_Schema_validate.py
import unittest
from function_Schema_validate import PersonValidator

class TestPersonValidator(unittest.TestCase):

    def test_valid_data(self):
        validator = PersonValidator()
        result = validator.validate("Alice", 30, "New York")
        self.assertEqual(result, "Valid data: name = Alice, age = 30, city = New York")

    def test_invalid_age(self):
        validator = PersonValidator()
        result = validator.validate("Alice", 15, "New York")
        self.assertEqual(result, "Invalid data: One or more fields are incorrect")

    def test_invalid_city(self):
        validator = PersonValidator()
        result = validator.validate("Bob", "25", 12345)
        self.assertEqual(result, "Invalid data: One or more fields are incorrect")

    def test_missing_city(self):
        validator = PersonValidator()
        result = validator.validate("Charlie", "28", "")
        self.assertEqual(result, "Invalid data: One or more fields are incorrect")

    def test_invalid_name(self):
        validator = PersonValidator()
        result = validator.validate("", "45", "London")
        self.assertEqual(result, "Invalid data: One or more fields are incorrect")

if __name__ == "__main__":
    unittest.main()
